web-admin/src/features/projects/github/ProjectGithubConnection.svelte

```diff
@@ -0,0 +1,128 @@
+<script lang="ts">
+  import ConnectToGithubConfirmDialog from "@rilldata/web-admin/features/projects/github/ConnectToGithubConfirmDialog.svelte";
+  import { GithubConnection } from "@rilldata/web-admin/features/projects/github/GithubConnection";
+  import GithubRepoSelectionDialog from "@rilldata/web-admin/features/projects/github/GithubRepoSelectionDialog.svelte";
+  import { Button } from "@rilldata/web-common/components/button";
+  import EditIcon from "@rilldata/web-common/components/icons/EditIcon.svelte";
+  import Github from "@rilldata/web-common/components/icons/Github.svelte";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { createAdminServiceGetProject } from "@rilldata/web-admin/client";
+  import { useDashboardsLastUpdated } from "@rilldata/web-admin/features/dashboards/listing/selectors";
+  import { getRepoNameFromGithubUrl } from "@rilldata/web-admin/features/projects/github/github-utils";
+
+  export let organization: string;
+  export let project: string;
+
+  $: proj = createAdminServiceGetProject(organization, project);
+  $: isGithubConnected = !!$proj.data?.project?.githubUrl;
+  $: repoName =
+    $proj.data?.project?.githubUrl &&
+    getRepoNameFromGithubUrl($proj.data.project.githubUrl);
+  $: subpath = $proj.data?.project?.subpath;
+  $: githubLastSynced = useDashboardsLastUpdated(
+    $runtime.instanceId,
+    organization,
+    project,
+  );
+
+  let confirmDialogOpen = false;
+  let githubSelectionOpen = false;
+
+  const githubConnection = new GithubConnection(() => {
+    githubSelectionOpen = true;
+  });
+  const userStatus = githubConnection.userStatus;
+
+  function connectToGithub() {
+    confirmDialogOpen = true;
+  }
+
+  function confirmConnectToGithub() {
+    confirmDialogOpen = false;
+    void githubConnection.check();
```

Comment:
Here as well, I had to read the function implementation to understand what was going on. What about just 
```
const githubUserStatusQuery = createAdminServiceGetGithubUserStatus();

...

window.open(
  $githubUserStatusQuery.data?.grantAccessUrl,
  "_blank",
)
```